gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsTimeStatistic.java

```diff
@@ -0,0 +1,104 @@
+package com.google.cloud.hadoop.fs.gcs;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Maps;
+import java.util.EnumSet;
+import org.apache.hadoop.classification.InterfaceStability;
+
+/** Time Statistics which are collected in GCS. */
+@InterfaceStability.Unstable
+public enum GhfsTimeStatistic {
+  /** Total Time for open(). */
+  OPEN("timeElapsedOpenMusec", "total time required for open in µ seconds", 0),
+
+  /** Total Time for create(). */
+  CREATE("timeElapsedCreateMusec", "total time required for create in µ seconds", 0),
+
+  /** Total Time for rename(). */
+  RENAME("timeElapsedRenameMusec", "total time required for rename in µ seconds", 0),
+
+  /** Total Time for delete(). */
+  DELETE("timeElapsedDeleteMusec", "total time required for delete in µ seconds", 0),
+
+  /** Total Time for close(). */
+  CLOSE("timeElapsedCloseMusec", "total time required for close in µ seconds", 0),
+
+  /** Total Time for hflush(). */
+  HFLUSH("timeElapsedHflushMusec", "total time required for hflush in µ seconds", 0),
+
+  /** Total Time for hsync(). */
+  HSYNC("timeElapsedHsyncMusec", "total time required for hsync in µ seconds", 0),
+
+  /** Total Time for seek(). */
+  SEEK("timeElapsedSeekMusec", "total time required for seek in µ seconds", 0);
+
+  public static final ImmutableSet<GhfsTimeStatistic> VALUES =
```

Comment:
why not https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/ImmutableEnumSet.java?

Reply:
I was referring to GhfsStatistic class and implemented the same way. Changing this to ImmutableEnumSet will have no impact in the connector performance (as per my understanding) as it's role to list out enum object's symbols in the plugin. In connector, it is not called anywhere.